apis/v1alpha2/gateway_types.go

```diff
@@ -397,17 +397,24 @@ const (
 
 	// This reason is used with the "ResolvedRefs" condition when the
 	// Listener has a TLS configuration with at least one TLS CertificateRef
-	// that is invalid or cannot be resolved.
+	// that is invalid or does not exist.
+	// A CertificateRef is considered invalid when it refers to a nonexistent
+	// or unsupported resource or kind, or when the data within that resource
+	// is malformed.
+	// This reason must be used only when the reference is granted, which means
+	// that if the referenced secret is invalid or does not exist, but the
+	// reference is not allowed because of the lack of a ReferenceGrant, the
+	// reason RefNotPermitted must be used.
```

Comment:
If this suggestion is accepted, it should be applied to the v1beta1 APIs too.

```suggestion
	// This reason must be used only when the reference is allowed, either by
	// referencing an object in the same namespace as the Gateway, or when
	// a cross-namespace reference has been explicitly allowed by a ReferenceGrant.
	// If the reference is not allowed, the reason RefNotPermitted must be used
	// instead.
```

Reply:
done